pkgs/development/python-modules/libknot/default.nix

```
@@ -0,0 +1,43 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+
+# build-system
+, hatchling
+
+# native dependencies
+, knot-dns
+}:
+
+buildPythonPackage rec {
+  pname = "libknot";
+  version = "3.3.1";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-zoaDzqTj31wHsGCz/Qw/2J0bQ9CtkWqlSHb4y+JV8U8=";
+  };
+
+  postPatch = ''
+    substituteInPlace libknot/__init__.py \
+      --replace "libknot%s.dylib" "${knot-dns.out}/lib/libknot%s.dylib" \
+      --replace "libknot.so%s" "${knot-dns.out}/lib/libknot.so%s"
+  '';
+
+  nativeBuildInputs = [
+    hatchling
+  ];
+
+  pythonImportsCheck = [
+    "libknot"
+  ];
+
+  meta = with lib; {
+    description = "Python bindings for libknot";
+    homepage = "https://pypi.org/project/libknot/";
```

```suggestion
    homepage = "https://gitlab.nic.cz/knot/knot-dns/-/tree/master/python/libknot";
```
this is probably more helpful